server/src/test/java/pl/kejbi/tin/SocketTest.groovy

```diff
@@ -0,0 +1,123 @@
+package pl.kejbi.tin
+
+import org.slf4j.Logger
+import pl.kejbi.tin.socket.Client
+import pl.kejbi.tin.socket.Server
+import pl.kejbi.tin.socket.builder.ServerSocketBuilder
+import pl.kejbi.tin.socket.builder.SocketBuilder
+import spock.lang.Specification
+
+class SocketTest extends Specification {
+
+    Socket socket;
+    ServerSocket serverSocket;
+
+    def createServerSocket() {
+        return new ServerSocketBuilder().createServerSocket().build()
+    }
+
+    def createClientSocket() {
+        return new SocketBuilder().createSocket().build()
+    }
+
+    def cleanup() {
+        if(socket != null) {
+            if(!socket.isClosed()) {
+                socket.close()
+            }
+            socket = null
+        }
+        if(serverSocket != null) {
+            if(!serverSocket.isClosed()) {
+                serverSocket.close()
+            }
+            serverSocket = null
+        }
+    }
+
+    def "should create server socket"() {
+        given:
+        ServerSocketBuilder serverSocketBuilder = new ServerSocketBuilder()
+        serverSocketBuilder.logger = Mock(Logger)
+
+        when:
+        serverSocket = serverSocketBuilder.createServerSocket().withPort(50000).build()
+
+        then:
+        0 * serverSocketBuilder.logger.error(!null)
+        serverSocket.getLocalPort() == 50000
+    }
+
+    def "should create socket"() {
```

Comment:
tu podobnie ja bym napisał `should create client socket with correct port and hostname`

Reply:
ok